### PR TITLE
Async storage

### PR DESCRIPTION
**Pull request by diozenio**
# Async Storage
- Async Storage implementation 

# Proposed Changes 
- update routers for support auth and app routes
- implement Async storage to store users info and the content of the publications 
- add signup and signup forms and their validations 
- add a form for write the publication content
- add context api 

# Additional Info
- List of publications not implemented 
- React Native Paper has not been used yet

# Screenshots 

- [Authentication Router](https://user-images.githubusercontent.com/68879723/178870845-b9338141-c30e-4569-976f-145625d874f0.png)
- [SignIn](https://user-images.githubusercontent.com/68879723/178870845-b9338141-c30e-4569-976f-145625d874f0.png)
- [SignUp](https://user-images.githubusercontent.com/68879723/178871017-0930447c-ae7f-405e-95cc-85a4c81b61ed.png)
- [Crete Message Screen](https://user-images.githubusercontent.com/68879723/178871084-5eab8931-d9aa-498c-a1df-5ff91321aaae.png)

### CODE REVIEW

File: @types/swal.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare module 'react-native-sweet-alert' {
+  const swal: {
+    showAlertWithOptions: (options: any) => void;
```
Comment (Manuel-Antunes) [inline @types/swal.d.ts:3]: Delete this swal declaration file

File: package.json [hunk 2]
```diff
@@ -22,6 +23,8 @@
     "react-native": "0.68.2",
     "react-native-safe-area-context": "4.2.4",
     "react-native-screens": "~3.11.1",
+    "react-native-sweet-alert": "^3.5.0",
```
Comment (Manuel-Antunes) [inline package.json:26]:
```suggestion
```

File: src/contexts/AuthContext.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import React from "react";
```
Comment (Manuel-Antunes) [inline src/contexts/AuthContext.tsx:1]: Split the typings, the hooks and the context

File: src/routes/auth.routes/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import SignIn from "../../screens/auth/SignIn";
+import SignUp from "../../screens/auth/SignUp";
+import Authentication from "../../screens/auth/Authentication";
+
+const Auth = createNativeStackNavigator();
+
+const AuthRoutes = () => {
+  return (
+    <Auth.Navigator screenOptions={{ headerShown: false }}>
+      <Auth.Screen name="Authentication" component={Authentication} />
```
Comment (Manuel-Antunes) [inline src/routes/auth.routes/index.tsx:12]: Change this page name to landing page

**Review by Manuel-Antunes — CHANGES_REQUESTED** [review]
Solve my issues then request my review again

File: package.json (hunk 2 above)
Comment (diozenio) [reply package.json]: The dependency has been removed

File: src/routes/auth.routes/index.tsx (hunk 4 above)
Comment (diozenio) [reply src/routes/auth.routes/index.tsx]: Page name changed to LandingPage

File: src/contexts/AuthContext.tsx (hunk 3 above)
Comment (diozenio) [reply src/contexts/AuthContext.tsx]: Typings, hooks and the context have been splitted

**Review by Manuel-Antunes — APPROVED** [review]
OK!

**Pull request merged**